stats/kafka/config.go

```diff
@@ -0,0 +1,121 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package kafka
+
+import (
+	"time"
+
+	"github.com/kubernetes/helm/pkg/strvals"
+	"github.com/loadimpact/k6/lib/types"
+	"github.com/loadimpact/k6/stats/influxdb"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/guregu/null.v3"
+)
+
+// Config is the config for the kafka collector
+type Config struct {
+	// Connection.
+	Brokers []string `json:"brokers" envconfig:"KAFKA_BROKERS"`
+
+	// Samples.
+	Topic        null.String        `json:"topic" envconfig:"KAFKA_TOPIC"`
+	Format       null.String        `json:"format" envconfig:"KAFKA_FORMAT"`
+	PushInterval types.NullDuration `json:"push_interval" envconfig:"KAFKA_PUSH_INTERVAL"`
+
+	InfluxDBConfig influxdb.Config `json:"influxdb"`
+}
+
+// config is a duplicate of ConfigFields as we can not mapstructure.Decode into
+// null types so we duplicate the struct with primitive types to Decode into
+type config struct {
+	Brokers      []string `json:"brokers" mapstructure:"brokers" envconfig:"KAFKA_BROKERS"`
+	Topic        string   `json:"topic" mapstructure:"topic" envconfig:"KAFKA_TOPIC"`
+	Format       string   `json:"format" mapstructure:"format" envconfig:"KAFKA_FORMAT"`
+	PushInterval string   `json:"push_interval" mapstructure:"push_interval" envconfig:"KAFKA_PUSH_INTERVAL"`
+
+	InfluxDBConfig influxdb.Config `json:"influxdb" mapstructure:"influxdb"`
+}
+
+// NewConfig creates a new Config instance with default values for some fields.
+func NewConfig() Config {
+	return Config{
+		Format:       null.StringFrom("json"),
+		PushInterval: types.NullDurationFrom(1 * time.Second),
+	}
+}
+
+func (c Config) Apply(cfg Config) Config {
+	if len(cfg.Brokers) > 0 {
+		c.Brokers = cfg.Brokers
+	}
+	if cfg.Format.Valid {
+		c.Format = cfg.Format
+	}
+	if cfg.Topic.Valid {
+		c.Topic = cfg.Topic
+	}
+	if cfg.PushInterval.Valid {
+		c.PushInterval = cfg.PushInterval
+	}
+	return c
+}
+
+// ParseArg takes an arg string and converts it to a config
+func ParseArg(arg string) (Config, error) {
```

Comment:
I don't quite like how complex the configuration parsing is turning out to be, but I think that's more of a reflection of the limitations we have with the current configuration approach in k6 and the libraries we use (`nil`-able types, `mapstructure`, `strvals`, `envconfig`, etc. don't play very nice together). So it's fine for now, but definitely something I'd like to revisit in the future.